src/engine/engine_bf.c

```diff
@@ -96,49 +106,40 @@ static int _ocf_backfill_do(struct ocf_request *req)
 		return 0;
 	}
 
+	seek = req->addr % cache_line_size;
+	last_chunk_size = (req->addr + req->bytes) % cache_line_size;
+	skip = (cache_line_size - last_chunk_size) % cache_line_size;
```

Comment:
This math is very confusing. The `last_chunk_size` is not really last chunk size, because it can be 0 if last chunk is aligned to cache line size, and then to account for that, the `skip` is "corrected" with this surprising modulo.